### DISCUSSION TITLE
tutorials/graphos-growing-supergraph/05-reference-entity

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#51] [course feedback]
# tutorials/graphos-growing-supergraph/05-reference-entity

https://www.apollographql.com/tutorials/graphos-growing-supergraph/05-reference-entity

**Comment by OldBoneBro** [comment]
Hello! :)

Anyone who finished the "GraphOS: Shipping your supergraph" course would be confused seeing the `text` field in the query, since we've deprecated and removed it from our schema. Maybe we should've changed it to `detailedDescription` which works a treat?

**Reply by mabuyo** [reply]
Oh good catch, thank you, I appreciate the callout! I'll add this to my list ✍🏽

**Comment by OldBoneBro** [comment]
>Next, inside the body of the function, we're using the dataSource method `getRecipeCookware` to retrieve a list of the recipe's cookware. This method was already implemented for us

But where? I've checked the 'src/datasources/recipes-api.js` and almost all the rest `src/` files and haven't found the  `getRecipeCookware` method. I guess that's one of the reasons why I'm getting such an error while testing the query with local router
```
errors": [
    {
      "message": "dataSources.recipesAPI.getRecipeCookware is not a function",
      "locations": [
        {
          "line": 1,
          "column": 112
        }
      ],
      "path": [
        "recipe",
        "cookware"
      ],
      ...

```
Eather the method doesn't exist or I miss something important
Help would be appreciated :)

**Reply by mabuyo** [reply]
Hey @OldBoneBro ! If you completed the previous two courses before May 24, then your copy of the Recipes API codebase is probably outdated. Could you take a look at [Lesson 1](https://www.apollographql.com/tutorials/graphos-growing-supergraph/01-growing-our-supergraph) under the **Project setup** section, open up the **Updates to the `recipes` subgraph** section and follow those instructions? There you'll find the snippet for `getRecipeCookware` function as well as updates to the JSON file. Let me know if that helps!

**Reply by OldBoneBro** [reply]
It sure did
Thanks a lot! :)
